packages/amplify-cli/src/domain/amplify-toolkit.ts

```diff
@@ -442,4 +444,21 @@ export class AmplifyToolkit {
       require(path.join(this._amplifyHelpersDirPath, 'get-imported-auth-properties')).getImportedAuthProperties;
     return this._getImportedAuthProperties;
   }
+
+  get runCleanUpTasks(): any {
+    this._runCleanUpTasks = this._runCleanUpTasks || require(path.join(this._amplifyHelpersDirPath, 'run-cleanup-tasks')).runCleanUpTasks;
```

Comment:
I know other functions in this class dynamically require method implementations, but here it seems like it would be much cleaner to just define the implementation here. That way it could directly reference the _cleanUpTasks property so you could run `context.amplify.runCleanUpTasks()` rather than the awkward `context.amplify.runCleanUpTasks(context)`. In that case you also wouldn't have to expose `getCleanUpTasks()` which could potentially be abused to remove or modify tasks

Reply:
That's a fair point we can keep the function within the toolkit. As for passing the context, I do agree we can improve that though there are other functions within the toolkit that also pass the context as an argument. If we want to improve that we should consider that as a broad change to handle overall within the Context.